### PR TITLE
Consolidate RazorSdkDirectory logic

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Addresses https://github.com/dotnet/aspnetcore/issues/22292

### CODE REVIEW

**Comment by JunTaoLuo** [comment]
Though this works, it's certainly not pretty.

File: eng/targets/RazorSDKDirectoryOverride.props [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<Project>
+  <PropertyGroup>
+    <RazorSdkDirectoryRoot Condition="'$(RazorSdkDirectoryRootOverride)' != ''">$(RazorSdkDirectoryRootOverride)</RazorSdkDirectoryRoot>
```
Comment (JunTaoLuo) [inline eng/targets/RazorSDKDirectoryOverride.props:3]: Allow tests to override this. The reason this is needed is that some tests build templates with the Debug configuration. However, given that the SDK along with tests are all built in the Release configuration, we need to override the property to ensure the SDK is resolved correctly.

File: eng/targets/RazorSDKDirectoryOverride.props [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+    <RazorSdkDirectoryRoot Condition="'$(RazorSdkDirectoryRootOverride)' == ''">$(ArtifactsBinDir)Microsoft.NET.Sdk.Razor\$(Configuration)\sdk-output\</RazorSdkDirectoryRoot>
```
Comment (wtgodbe) [inline eng/targets/RazorSDKDirectoryOverride.props:4]: I think the general pattern would be:

```<RazorSdkDirectoryRoot Condition="'$(RazorSdkDirectoryRoot)' == ''">```

File: src/ProjectTemplates/BlazorTemplates.Tests/Infrastructure/TemplateTests.props.in [hunk 3]
```diff
@@ -34,9 +34,8 @@
     several versions older than latest. To avoid a cyclical dependency, this package reference is added to override the bundled version.
     Since this is a project reference, we must explicitly import the props file and also specify the output location of the SDK directory.
   -->
-  <Import Project="${RepoRoot}src\Razor\Microsoft.NET.Sdk.Razor\src\build\netstandard2.0\Microsoft.NET.Sdk.Razor.props" Condition="'$(UsingMicrosoftNETSdkWeb)' == 'true' OR '$(RazorSdkCurrentVersionProps)' != ''" />
-
-  <PropertyGroup>
-    <RazorSdkDirectoryRoot>${ArtifactsBinDir}Microsoft.NET.Sdk.Razor\${Configuration}\sdk-output\</RazorSdkDirectoryRoot>
-  </PropertyGroup>
+  <ImportGroup Condition="'$(UsingMicrosoftNETSdkWeb)' == 'true' OR '$(RazorSdkCurrentVersionProps)' != ''">
```
Comment (wtgodbe) [inline src/ProjectTemplates/BlazorTemplates.Tests/Infrastructure/TemplateTests.props.in:37]: Huh, I didn't know `ImportGroup` was a thing. cool

**Review by wtgodbe — APPROVED** [review]

File: eng/targets/RazorSDKDirectoryOverride.props (hunk 2 above)
Comment (JunTaoLuo) [reply eng/targets/RazorSDKDirectoryOverride.props:4]: Not quite, I want to be able to override the value with `RazorSdkDirectoryRootOverride` if it is set.

Comment (wtgodbe) [reply eng/targets/RazorSDKDirectoryOverride.props:4]: I meant just for the second line - if `RazorSdkDirectoryRootOverride` is empty, `RazorSdkDirectoryRoot` will be empty on line 4. Using the name of the property you're setting indicates more that you're setting it's default value, if it doesn't already have a value, e.g.

https://github.com/dotnet/aspnetcore/blob/8c1bf1f1a31f852880c16575bca62507b75112bf/eng/targets/ResolveReferences.targets#L49-L57

File: eng/targets/RazorSDKDirectoryOverride.props (hunk 1 above)
Comment (dougbu) [reply eng/targets/RazorSDKDirectoryOverride.props:3]: The `Condition` isn't useful here. Just set `$(RazorSdkDirectoryRoot)` to `$(RazorSdkDirectoryRootOverride)`.

Alternatively leave this line out entirely and set `$(RazorSdkDirectoryRoot)` where necessary. Right now, it looks like `$(RazorSdkDirectoryRootOverride)` is set as a global property when overridden anyhow.

File: src/ProjectTemplates/BlazorTemplates.Tests/BlazorTemplates.Tests.csproj [hunk 4]
```diff
@@ -68,6 +68,10 @@
       <_Parameter2>$(TargetFramework)</_Parameter2>
       <_Parameter3></_Parameter3>
     </AssemblyAttribute>
+    <AssemblyAttribute Include="System.Reflection.AssemblyMetadataAttribute">
+      <_Parameter1>Test.RazorSdkDirectoryRootOverride</_Parameter1>
+      <_Parameter2>$(RazorSdkDirectoryRoot)</_Parameter2>
```
Comment (dougbu) [inline src/ProjectTemplates/BlazorTemplates.Tests/BlazorTemplates.Tests.csproj:73]: The more I see of the `RazorSdkDirectoryRootOverride` name, the less I think it's needed

File: eng/targets/RazorSDKDirectoryOverride.props [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<Project>
```
Comment (dougbu) [inline eng/targets/RazorSDKDirectoryOverride.props:1]: Why not just `GetRazorSDKDirectory.props`❔

**Review by dougbu — CHANGES_REQUESTED** [review]
Naming is hard

**Review by dougbu — APPROVED** [review]

**Pull request merged**